# **Investigating 3D expansion**

> Expand a little on the main result (which is that adding genes whose proms have contacts with regions with DM CpGs improves the deconvolution performance. We could maybe have a look at which genes are added and speculate on why?

> 3D epigenomics - better justify in text, look at results of 3dprometâ€¦

<details>
  <summary>Genes added to BPRNACan for BPRNACanProMet</summary>

> ACLY, AGGF1, AKT1, ANK1, ANKS3, AP3M2, AP5S1, ARHGAP9, ARHGDIA, ARHGEF39, ARSG, ATG4B, ATG4D, ATP2A3, BCL2L2, BEND3, BIN2, BRAT1, C11orf42, C12orf57, C16orf71, C7orf26, C7orf50, CABLES2, CALM2, CAP2, CAPG, CAPN15, CARD10, CBX2, CCDC106, CCDC107, CD320, CD4, CD82, CDC42EP4, CDK11B, CDKN2D, CDSN, CEBPE, CENPM, CGA, CIB1, CITED4, CLK4, CLPTM1, COMMD8, COPZ1, CORO1A, CRNKL1, CSNK2B, CTSD, CTSZ, CUL9, CYC1, CYTH1, DCPS, DCUN1D2, DDAH2, DDR1, DHODH, DHTKD1, DIABLO, DLX2, DNASE1L2, DNMBP, DOK3, DPCD, DTNB, DUS2, E4F1, EHMT2, ENTPD2, EPHX3, ERI1, ERI2, ESPN, ETHE1, EXOSC4, FAM160A2, FAM167B, FASN, FASTK, FOXK2, FYCO1, FZR1, GABRB1, GCGR, GGT1, GLB1L, GLS2, GMEB2, GOLGA2, GPAA1, GPANK1, GPD2, GSG1, HAND2, HELZ2, HIST1H2AL, HLA-E, HSPA9, HSPB6, ICAM3, IGFLR1, IKZF4, IL18BP, INPP5J, IQCE, ITGAE, ITPKC, KLHDC4, KLHL6, LBX2, LIN37, LMBR1L, LOXL3, LRRC32, LRTOMT, LTC4S, LY6G6C, LY6G6D, LYL1, LYNX1, LYPD2, MAB21L3, MADD, MAF1, MAFK, MAML1, MAP4K2, MAP4K4, MARS, MEN1, MEPCE, MGAT1, MGAT4B, MIB2, MIP, MLF2, MLST8, MMP23B, MTCH1, MTG1, MTHFD1, MTMR14, MUC4, MVD, MYB, MYBPC3, MYO1A, NAA40, NABP2, NACC1, NCAPD2, NCK2, NCLN, NDUFS8, NFKBIB, NKAPL, NLRP6, NOL9, NOP10, NPDC1, NPEPL1, NPFF, NPRL2, NR1H3, NR2F6, NRL, NSFL1C, NUMA1, OR52W1, OS9, OSR1, P2RX4, P4HB, PACS2, PAK2, PAOX, PAX8, PCDHGB6, PCK2, PDE6A, PDIA5, PDLIM7, PENK, PHC3, PHKG1, PHLPP2, PIEZO1, PIGX, PILRB, PLA2G15, POLL, POP7, POU4F2, PPP1CA, PPP1R27, PRKAR2A, PRLHR, PROZ, PSENEN, PSORS1C1, PTPN22, PTPN6, PUF60, RAD9A, RASSF1, RAVER1, RBM23, RBM4B, RELB, RELT, RHOF, RIPPLY3, RNF121, RNF166, ROCK2, RPN1, RTEL1, S1PR4, SAMM50, SAR1A, SCRIB, SEMA4B, SERPINF2, SERPINH1, SETD1B, SGSM3, SGTA, SHARPIN, SHROOM1, SIAH1, SIM2, SIPA1L3, SIRT2, SLC15A3, SLC16A3, SLC16A6, SLC17A9, SLC19A1, SLC25A15, SLC48A1, SLC4A2, SLC7A14, SLC7A6, SMG6, SNX21, SPATA13, SPIDR, SQSTM1, SRCAP, SRR, SSH3, SSTR5, STEAP3, STK16, SUCLG2, SULT1A3, SUOX, SYCP2L, SYNJ2, SYS1, TARBP2, TBC1D20, TBX4, TCIRG1, TEX30, TFDP1, THAP4, THOP1, TLE6, TLX1, TLX2, TMCO3, TMEM132A, TMEM14B, TMEM184A, TMEM63C, TMUB1, TMUB2, TNF, TNKS1BP1, TNNC2, TPST2, TRA2B, TRMT1, TYK2, U2AF1L4, U2AF2, UBR5, VASN, VPS18, VPS9D1, VSX1, WDR87, WIPI1, WWP2, XXYLT1, ZBTB12, ZCWPW1, ZDHHC4, ZHX2, ZMYND10, ZNF132, ZNF205, ZNF276, ZNF500, ZNF641, ZNF668, ZNF74, ZSCAN10, ZSCAN18

</details>
<details>
  <summary>Genes added to BPRNACan for BPRNACan3DProMet (63) </summary>

> ACLY, ARHGDIA, ARSG, ATG4D, BCL2L2, BRAT1, C12orf57, C7orf50, CAPG, CBX2, CD4, CD82, CDKN2D, CLK4, COPZ1, CYTH1, DCUN1D2, DNMBP, ERI1, FASN, FOXK2, GCGR, GLS2, HELZ2, HIST1H2AL, HLA-E, HSPB6, ICAM3, IGFLR1, IQCE, KLHL6, LBX2, LIN37, LMBR1L, MAFK, MIP, MLF2, MTMR14, MYBPC3, NCK2, OSR1, P4HB, PPP1R27, PSENEN, PTPN6, RAVER1, RBM23, RPN1, SLC16A3, SLC16A6, SPATA13, SPIDR, SYNJ2, TLX2, TMCO3, TMEM184A, TNKS1BP1, TYK2, U2AF1L4, UBR5, WIPI1, ZNF641, ZNF668

</details>
<details>
  <summary>Genes in BPRNA not in BPRNACan anymore (307) </summary>

>  ABCG1, ABHD14B, ABI3, ADTRP, AGAP1, AIF1, AKR1C1, AKR1C3, ALOX5, ANKH, AP1S3, APP, ARHGEF11, ARHGEF4, ARRB1, ASIC3, ATF3, ATP2B4, ATP8B4, ATRN, B3GALT2, BAALC, BACE2, BCL11A, BEND7, BIVM, BMP6, BTBD19, BTNL9, C19orf48, C20orf27, C2orf88, CACNA1E, CADM4, CAMK1, CAMK2D, CAMKK1, CASS4, CCL20, CCL3, CCL3L1, CCR1, CCR9, CD101, CD248, CD300A, CD302, CD4, CD82, CD83, CD93, CHI3L2, CHRFAM7A, CLEC4E, COL13A1, COL4A3, COX6C, CPNE7, CRISPLD2, CRKL, CRTAP, CST3, CXCL2, CXCR5, CYBRD1, DBN1, DCAF12, DMXL2, DNAAF1, DNAH14, DOCK9, DPH1, DPH6, DPPA4, DSTN, ECHDC3, EFNA5, EGR1, EGR2, ELOVL6, ERBB2, EREG, ESAM, EVA1B, EVI5, FAM114A1, FAM171A1, FAR2, FCER1G, FCGR2A, FCGR3A, FCRLA, FEZ1, FGD6, FHL3, FNIP2, FRMPD3, FUT8, G0S2, GAL3ST4, GAS7, GCM1, GCNT2, GFPT2, GIMAP4, GINS2, GIPR, GNA12, GNA15, GNB5, GPR183, GPR82, GREM2, H1F0, HAAO, HAVCR1, HCK, HENMT1, HES1, HES4, HIST1H1T, HK3, HLA-DRA, HLA-DRB5, HMGB3, HMOX1, HS3ST3A1, ICAM1, ICOS, IFIT3, IGF2BP2, IGSF6, IL1R1, IMMP1L, IMMP2L, INSR, ITGAD, ITGAX, JSRP1, KALRN, KBTBD11, KCNK5, KDM1A, KIAA0408, KIAA1257, KIF21A, KLF4, KLHL3, KRT72, LAIR2, LDLR, LGALS3, LILRA4, LIPE, LMTK3, LRRC23, LRRC56, LTA4H, MAGEH1, MAML2, MAP3K7CL, MBLAC2, MCCC2, MCM5, MMAA, MMP28, MNDA, MPZL2, MRPL28, MRPS9, MRVI1, MYL9, MYO1B, MYO1C, MYO1E, N6AMT1, NAGA, NBEA, NBL1, NBPF14, NDC80, NDUFAF1, NEIL3, NLRP3, NOD2, NOG, NPDC1, NRGN, NSUN7, NT5DC3, OASL, OBSCN, OPLAH, OR13A1, ORM1, PALD1, PAOX, PARP15, PDCD1, PDE4D, PDE6H, PEBP1, PF4V1, PFKFB2, PFKM, PHPT1, PILRA, PKIG, PLAUR, PLCG2, PLEKHA7, PLXND1, PODN, PPFIBP1, PPT1, PRDM10, PRDM5, PRSS35, PSMG1, PTCH1, PTGDR2, PTGES, PXMP4, RAB27B, RAB30, RANBP17, RBM11, RGS18, RILP, RIMKLB, RNASE4, RP9, RPS6KL1, RRP1B, S100A8, SASH1, SDC3, SECTM1, SEMA4A, SERPINB2, SERPING1, SETBP1, SETD7, SH2B2, SH3BGRL2, SLC16A1, SLC16A5, SLC19A1, SLC22A23, SLC26A8, SLC35G2, SMARCAD1, SORBS2, SOX13, SOX5, SPATA18, SPHK1, SPN, SPON2, SPRY2, STARD9, STYK1, SUSD3, SVIL, SYN3, SYTL3, TBC1D8, TBXAS1, TCL1B, TCTN2, TGM3, THBS1, TIAM2, TIMM13, TIMP1, TM6SF1, TMEM171, TMOD2, TNFRSF10A, TNFRSF10D, TNFRSF18, TOP1MT, TRAPPC6A, TRIM58, TRMT1, TRPC1, TRPM6, TSPAN13, TSR1, TST, TTC21A, UPB1, WNT3, ZBTB32, ZBTB46, ZC2HC1A, ZDHHC1, ZG16B, ZMYM5, ZMYND19, ZNF287, ZNF385A, ZNF519, ZNF608, ZNF781, ZNHIT6, ZSCAN29  

</details>

<details>
  <summary>Genes lost in BPRNACan from BPRNA but restored in BPRNACan3DProMet (2) </summary>

> CD4 and CD82

</details>

CD82 is highly expressed in Macrophages meanwhile CD4 is predictive to CD4 Tcells but highly expressed in cancer cells too.

| Gene | B | cancer | CD4 | CD8 | M0 | M1 | M2 | Mono | Neu | NK |
|----|----|----|----|----|----|----|----|----|----|----|
**CD82** | 84.3326385930249 | 64.5127166749311 | 18.9039662777569 | 23.8211997177599 | 102.211008948327 | **258.391212325986** | **220.407036029912** | 29.4471317199325 | 101.043018616383 | 0.455706678016801 |
**CD4** | 0.330954816228237 | **22.4139343320294** | **11.0265972708576** | 0 | 3.15251007047522 | 1.01637602768557 | 0 | 5.042559199645 | 0.675410302863387 | 0

Based on the paper [DNA Methylation in T-Cell Development and Differentiation](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8048391/pdf/nihms-1690197.pdf), CD4 expression responsible for Tcell differentiation is regulated by DNA methylation.  
Tcells differentiation going from Double Negative (CD4- ; CD8-) into DP (CD4+ ; CD8+) and for CD4TCells SP (CD4+ ; CD8-). In CD8+Tcells, CD4 downregulation due to DNMTs and CD4 enhanced expression in CD4+Tcells due to TETs.

***

### *To do*

1. Genes connection is cell type specific
  1. Associate cpgs to cell type
  2. Make a function that chose a cell type and find the connected genes
  3. Find that the majority of genes connected to macrophages are differentially expressed for macrophages.
  4. *Pray*
2. Cell type specific cpgs are located in hubs
  1. Make network pchic and cpgs cell type specific
  2. Look at macrophages
3. Make a pchic network of BPRNACan genes + Newly added Genes in another color
4. Do GO analysis on genes/cell types
5. Find some cluster of CpGs in IGV
6. Pray that one of the result is relevant and say hallelujah!

7.
  1. Make a cell specific network
  2. Add genes specific from 3D
  3. Add sig cpgs specific



***

## **I. Gene connection is cell type specific** *I hope*

### 1. Associate cpgs to cell type

Number of CpGs associated with each cell type

| B | cancer | CD4 | CD8 | M0 | M1 | M2 | Mono | Neu | NK | normal | wb | Total |
|----|----|----|----|----|----|----|----|----|----|----|----|----|
|954|1312|1301|1293|990|984|941|1038|969|1219|187|112|1896|

### 2. Association Genes with cell type using cpgs connections

#### B cells

<details>
  <summary>Genes with sigCpGs in promoter and connected to sigCpGs in enhancer through cell type specific chromatin contact</summary>

CD4 PRMT5-AS1 RBM23 ARSG SLC16A6 CBX2 SLA2 FAM78A


</details>

#### Cancer cells

<details>
  <summary>Genes with sigCpGs in promoter and connected to sigCpGs in enhancer through cell type specific chromatin contact</summary>

RNU7-163P, CD82, RP11-58K22.4, RP11-58K22.5, MYBPC3, AC090559.2, SPI1, TNKS1BP1, ADRBK1, ANKRD13D, ETS1, CD4, C12orf57, PTPN6, U47924.31, RP11-370I10.6, ZNF641, LMBR1L, GLT1D1, SPATA13-AS1, SPATA13, PRMT5-AS1, RBM23, BCL2L2, BCL2L2-PABPN1, AC135050.5, ZNF668, ACLY, TTC25, ARSG, SLC16A6, WIPI1, CYTH1, CBX2, AC174470.1, FAM195B, GCGR, PPP1R27, ARHGDIA, P4HB, RP11-498C9.3, CCDC57, FASN, TEX19, SLC14A1, CDKN2D, FAM129C, AC008397.1, LRRC25, AC002398.11, AC002398.9, AD000671.6, IGFLR1, LIN37, PSENEN, U2AF1L4, AC002398.12, C19orf55, HSPB6, OSR1, AC005041.17, LBX2-AS1, LBX2, RP11-523H20.3, TLX2, NCK2, AC009505.2, SLA2, HELZ2, RP11-188P20.3, SNRK-AS1, KLHL6, CTD-3080P12.3, CTC-436P18.1, SMIM15, CLK4, HIST1H2AL, HIST1H2BPS2, HCG14, HLA-E, AC073957.15, C7orf50, AC093734.1, MAFK, TMEM184A, BRAT1, IQCE, CEBPD, SPIDR, PVT1, FAM78A

</details>

#### CD4 Tcells

<details>
  <summary>Genes with sigCpGs in promoter and connected to sigCpGs in enhancer through cell type specific chromatin contact</summary>

TNKS1BP1, LAG3, MLF2, PTMS, CD4, C12orf57, PTPN6, U47924.31, LMBR1L, SPATA13-AS1, SPATA13, AC108004.3, ARSG, SLC16A6, WIPI1, CCDC57, FASN, RP13-516M14.2, SLC16A3, TEX19, AC009505.2, HELZ2, KLHL6, HLA-E, AC093734.1, MAFK, TMEM184A, FAM78A

</details>

#### Macrophages

<details>
  <summary> Genes with sigCpGs in promoter and connected to sigCpGs in enhancer through cell type specific chromatin contact </summary>

LMBR1L COPZ1 NFE2 PRMT5-AS1 RBM23 CYTH1 AC174470.1 FAM195B GCGR PPP1R27 ARHGDIA P4HB RP11-498C9.3 CCDC57 FASN SLC14A1 CAPG SH2D6 KLHL6 HCG14 BRAT1 IQCE KB-1980E6.2 UBR5 FAM78A  

[Link](https://docs.google.com/spreadsheets/d/1RUWdtWDLxlgH2cW-ockqFTFZjNfIRkEZIhYL-B4YWMM/edit?usp=sharing)

</details>


#### Monocyte

<details>
  <summary>Genes with sigCpGs in promoter and connected to sigCpGs in enhancer through cell type specific chromatin contact</summary>

LMBR1L, PRMT5-AS1, RBM23, CCDC57, FASN, CLK4, BRAT1, IQCE

</details>
